datafusion/sql/src/unparser/mod.rs

```diff
@@ -105,13 +109,33 @@ impl<'a> Unparser<'a> {
         self.pretty = pretty;
         self
     }
+
+    /// Add a custom unparser for user defined logical nodes
+    ///
+    /// DataFusion allows user to define custom logical nodes. This method allows to add custom child unparsers for these nodes.
+    /// Implementation of [`UserDefinedLogicalNodeUnparser`] can be added to the root unparser to handle custom logical nodes.
+    ///
+    /// The child unparsers are called iteratively.
+    /// There are two methods in [`Unparser`] will be called:
+    /// - `extension_to_statement`: This method is called when the custom logical node is a custom statement.
+    ///     If multiple child unparsers return a non-None value, the last unparsing result will be returned.
+    /// - `extension_to_sql`: This method is called when the custom logical node is part of a statement.
+    ///    If multiple child unparsers are registered for the same custom logical node, all of them will be called in order.
+    pub fn with_udlp_unparsers(
```

Comment:
Not a fan of this name - `udlp` takes effort to understand what it means. How about renaming `udlp_*` to `extension_*`? i.e. `with_extension_unparsers`. It conveys the same meaning in an easier to understand way.

Reply:
Sounds great. I'll rename it.